### PR TITLE
gorry fix 20170215

### PR DESCRIPTION
**Pull request by gorry**
Fix some Javascript Library code bugs found in my personal project.

### CODE REVIEW

File: src/library_sdl.js [hunk 1]
```diff
@@ -1101,6 +1101,7 @@ var LibrarySDL = {
         audio.webAudioNode['onended'] = function() { audio['onended'](); } // For <media> element compatibility, route the onended signal to the instance.
 
         audio.webAudioPannerNode = SDL.audioContext['createPanner']();
+        audio.webAudioPannerNode["setPosition"](0, 0, -.5);
```
Comment (kripken) [inline src/library_sdl.js]: please add a comment explaining why this is needed.

also, why `-.5` and not `0` for the last one?

File: src/library_sdl.js [hunk 2]
```diff
@@ -1225,6 +1226,8 @@ var LibrarySDL = {
       for (var joystick in SDL.lastJoystickState) {
         var state = SDL.getGamepad(joystick - 1);
         var prevState = SDL.lastJoystickState[joystick];
+        // PATCHED: If joystick was removed, state returns null.
+        if (typeof state == 'undefined') return;
```
Comment (kripken) [inline src/library_sdl.js]: please use `===` (no change in behavior, but it's our convention everywhere)

**Comment by kripken** [comment]
Thanks, nice fixes :)

Please also add yourself to AUTHORS.

**Comment by kripken** [comment]
Merged to incoming. Thanks!

**Comment by gorry** [comment]
I found the new problem in detaching joystick, to #4962.

**Pull request closed without merging**